### PR TITLE
Post editor: Make the Post Content block available as a child of the Query block

### PR DESCRIPTION
**Pull request by glendaviesnz**
## What?
Only removes the Post Content block from the post editor if the parent block is not the Query block.

## Why?
Possible fix for [issue noted here](https://github.com/WordPress/gutenberg/pull/50620#issuecomment-1586157695). 

## How?
Before setting the `canInsert` flag check the root block of the currently selected block.

## Testing Instructions

- In the post editor check that the Post Content block is not available
- Add a Query block and when the Query block is the parent check that the Post Content block is available

## Screenshots or screencast 

Before:
Screenshot 2023-06-12 at 1 30 24 PM

After:
Screenshot 2023-06-12 at 1 29 05 PM

### CODE REVIEW

**Comment by glendaviesnz** [comment]
I did explore adding this filter to the Post Content block init method, but couldn't find a way there of only limiting it to the post editor, it also removed it from all but Query block in the site editor.

File: packages/edit-post/src/index.js
Comment (nerrad) [inline packages/edit-post/src/index.js]:
```suggestion
		'removePostContentFromInserter',
```

**Comment by ntsekouras** [comment]
I pushed an [update](https://github.com/WordPress/gutenberg/pull/51405/commits/663a79b5e797cc164426c57883a83f7f8348c38d) to the logic to check for Query Loop parents. I'd appreciate some testing from others too 😄

**Review by nerrad — COMMENTED** [review]
I was in the midst of a review and caught the problem with the logic that Nik addressed. 

However, @ntsekouras, I think `getBlockParentsByBlockName` is already accessible to the filter callback so you could probably just use what's provided?

Also, this seems like logic that could be condensed into one filter callback and only executed when not editing templates (minor performance op).

**Comment by ntsekouras** [comment]
Good catch with `getBlockParentsByBlockName` being exposed. 

>Also, this seems like logic that could be condensed into one filter callback and only executed when not editing templates (minor performance op).

Agreed. I'd leave this to @glendaviesnz though. I wouldn't want to fully take over this PR. I was reviewing and played around with the logic before 😄

**Comment by glendaviesnz** [comment]
> I pushed an [update](https://github.com/WordPress/gutenberg/pull/51405/commits/663a79b5e797cc164426c57883a83f7f8348c38d) to the logic to check for Query Loop parents. I'd appreciate some testing from others too 😄

Ah, that makes more sense. Thanks @ntsekouras! 


> I wouldn't want to fully take over this PR

I love it when people fix my PRs while I am sleeping - it is like being in a real-life version of the [elves and the shoemaker fairy tale](https://en.wikipedia.org/wiki/The_Elves_and_the_Shoemaker) 😄 .

> Also, this seems like logic that could be condensed into one filter callback and only executed when not editing templates (minor performance op).

Thanks @nerrad, good thinking, have merged them.

File: packages/edit-post/src/index.js
Comment (talldan) [inline packages/edit-post/src/index.js]: In hindsight I wonder if it might be better to make a separate second call to `addFilter` for the post content block and leave the template part code in tact.

The reason is that some devs might use `removeFilter( 'blockEditor.__unstableCanInsertBlockType', 'removeTemplatePartsFromInserter' )`, if the name is changed it would stop that `removeFilter` call from working.

**Review by ntsekouras — APPROVED** [review]
LGTM, thank you Glen!

File: packages/edit-post/src/index.js
Comment (ntsekouras) [reply packages/edit-post/src/index.js]: 😄  I approved before seeing your comment. This makes sense.

Comment (nerrad) [reply packages/edit-post/src/index.js]: Oh ya, good thought.

Comment (glendaviesnz) [reply packages/edit-post/src/index.js]: Good spotting Dan, thanks, will update.

File: packages/edit-post/src/index.js
Comment (talldan) [inline packages/edit-post/src/index.js:93]: The comment above this on line 82 still mentions post content.

**Review by talldan — APPROVED** [review]
Works great, just one tiny nitpick about the comment that needs to be updated before merge.

File: packages/edit-post/src/index.js
Comment (glendaviesnz) [reply packages/edit-post/src/index.js:93]: rats, I was sure I removed that! Gone now.

**Comment by gaambo** [comment]
I‘m copying my comment here as well to maybe get a response:
Shouldnt that be possible with the [ancestor](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#ancestor) property when registering the block? Or am I missing something obvious here?

**Pull request merged**
